### ISSUE TITLE
Consider moving away from server sent events as not supported in IE - SignalR?

### ISSUE THREAD

**Issue by alexmackey** [sbarski/dashing.net#1]

**Comment by sbarski** [comment]
We could fork the solution and try SignalR. The good thing about SSE is that it's done over HTTP (i.e. doesn't require another protocol) and is very lightweight.

**Comment by rjk** [comment]
+1 for this. IE support would be great.

**Comment by serbrech** [comment]
I agree with @sbarski, SSE is a standard, it's only a shame that IE does not implement it.
Also, using Signalr would mean changing some of the front end (js communication), and the point here is that we can just reuse the original front end from shopify.
Just my 2 cents.

**Comment by alexmackey** [comment]
It would probably mean creating two server side broadcasting points but I wonder if it would be possible to implement an EventSource object that would provide polling/websockets if SSE were not available & allow reuse of existing shopify code.

**Comment by sbarski** [comment]
I think we should maintain SSE as our standard. I think we should keep this (canonical) version of dashing.net compatible with the original dashing and with the other implementations (in python and perl) that are currently being written.

For IE support we have a couple of options:
1. We can fork dashing.net and replace SSE with SignalR (sr-dashing.net?). If someone is very keen to do that and to maintain the fork please let me know. (Although I am still not sure it will be a completely worthwhile exercise.)
2. Alternatively we can explore an IE polyfill to fake an EventSource object via JavaScript. The Yaffle polyfill might do the trick although it has a few limitations (I think it completely replaces the EventSource object even in non-IE browsers - but I need to have a look at it to be sure). 

There might be other polyfills we could use or we could roll our own. This could be my weekend project but if anyone want to give it a shot - please feel free :)

**Comment by serbrech** [comment]
I think the polyfill option is be the best one. It respects the open-close principle (sort of). We don't change anything, we just extend when needed.

It's even an interesting addition to the original dashing project, I'm sure they would be happy to merge a PR that make them support IE (hum hum).

**Comment by pmackay** [comment]
Did this ever get any further? The demo doesnt currently seem to work in IE though.

**Comment by newman-xu** [comment]
How can we make it to support IE? Actually IE still has big share in browsers. We have to support it before we use this.
